File: uva_369.cpp
# include <iostream>
using namespace std;
int main(void){
    int m=0, n=0;
    while (cin >> m >> n){
        if(!m && !n){
            break;
        }
        long long int ans=1;
        int _n=n;
        if (n>m/2)
            _n=m-n;
        for (int i=0; i<_n; ++i){
            ans*=(m-i);
            ans/=(1+i);
        }
        cout << m << " things taken " << n << " at a time is " << ans << " exactly.\n";
    }
    return 0;
}

File: uva_369.py
while True:
    m, n = input().split()
    m, n = int(m), int(n)
    if not m and not n:
        break
    _n=n
    ans=1
    if n>m/2:
        _n=m-n
    for i in range(_n):
        ans*=(m-i)
        ans/=(1+i)
    print("{} things taken {} at a time is {} exactly.".format(m, n, int(ans)) )
